x/gov/router.go

```diff
@@ -0,0 +1,51 @@
+package gov
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/cosmos/cosmos-sdk/x/gov/proposal"
+)
+
+// Router is a map from string to proposal.Handler
+// copied and modified from baseapp/router.go
+type Router interface {
```

Comment:
This is now the 3rd implementation of a router (the other two are in BaseApp). I think it's finally time to abstract a router into `types/`.

Don't bother updating BaseApp rn, but we should at least add one to types.

Reply:
I will open a separate issue, we really need to abstract out the router but since the handler types are all different it is hard to do it without reflection. Will cover in another PR.

Reply:
Fair enough. Create a followup issue and add a reference comment here.

Reply:
https://github.com/cosmos/cosmos-sdk/issues/3976